stats/stats_linux.go

```diff
@@ -0,0 +1,159 @@
+package stats
+
+import (
+	"context"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+	"sync"
+	"time"
+)
+
+type Stats struct {
+	cpuCount int
+
+	loadAvg1Min  string
+	loadAvg5Min  string
+	loadAvg15Min string
+
+	memTotal     string
+	memFree      string
+	memAvailable string
+
+	mu sync.RWMutex
+}
+
+// Failures here aren't critical, so we just log them.
+func New(ctx context.Context) *Stats {
+	s := new(Stats)
+
+	count, err := getCPUCount()
+	if err != nil {
+		slog.Info("failed to get cpu count", "err", err)
+	}
+
+	s.cpuCount = count
+
+	go func() {
+		ticker := time.NewTicker(time.Second)
+		defer ticker.Stop()
+		for {
+			select {
+			case <-ticker.C:
+				s.tick()
+			case <-ctx.Done():
+			}
+		}
+	}()
+
+	return s
+
+}
+
+func (s *Stats) tick() {
+	s.mu.Lock()
```

Comment:
get all the values necessary _before_ the lock, then lock, and then set the `s.loadAvg1Min`, etc.